File: docs/breez-sdk/snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ListPayments.kt
package com.example.kotlinmpplib

import breez_sdk_spark.*
class ListPayments {
    suspend fun getPayment(client: BreezClient) {
        // ANCHOR: get-payment
        try {
            val paymentId = "<payment id>";
            val response = client.payments().get(GetPaymentRequest(paymentId))
            val payment = response.payment
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: get-payment
    }

    suspend fun listPayments(client: BreezClient) {
        // ANCHOR: list-payments
        try {
            val response = client.payments().list(ListPaymentsRequest())
            val payments = response.payments
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-payments
    }

    suspend fun listPaymentsFiltered(client: BreezClient) {
        // ANCHOR: list-payments-filtered
        try {
            // Filter by asset (Bitcoin or Token)
            val assetFilter = AssetFilter.Token(tokenIdentifier = "token_identifier_here")
            // To filter by Bitcoin instead:
            // val assetFilter = AssetFilter.Bitcoin

            val response = client.payments().list(
                ListPaymentsRequest(
                    // Filter by payment type
                    typeFilter = listOf(PaymentType.SEND, PaymentType.RECEIVE),
                    // Filter by status
                    statusFilter = listOf(PaymentStatus.COMPLETED),
                    assetFilter = assetFilter,
                    // Time range filters
                    fromTimestamp = 1704067200u, // Unix timestamp
                    toTimestamp = 1735689600u,   // Unix timestamp
                    // Pagination
                    offset = 0u,
                    limit = 50u,
                    // Sort order (true = oldest first, false = newest first)
                    sortAscending = false
                ))
            val payments = response.payments
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-payments-filtered
    }
}

File: docs/breez-sdk/snippets/react-native/list_payments.ts
import type { Payment, BreezSdk } from '@breeztech/breez-sdk-spark-react-native'
import { PaymentType, PaymentStatus, AssetFilter } from '@breeztech/breez-sdk-spark-react-native'

const exampleGetPayment = async (sdk: BreezSdk): Promise<Payment> => {
  // ANCHOR: get-payment
  const paymentId = '<payment id>'
  const response = await sdk.getPayment({
    paymentId
  })
  const payment = response.payment
  // ANCHOR_END: get-payment
  return payment
}

const exampleListPayments = async (sdk: BreezSdk): Promise<Payment[]> => {
  // ANCHOR: list-payments
  const response = await sdk.listPayments({
    typeFilter: undefined,
    statusFilter: undefined,
    assetFilter: undefined,
    fromTimestamp: undefined,
    toTimestamp: undefined,
    offset: undefined,
    limit: undefined,
    sortAscending: undefined,
    sparkHtlcStatusFilter: undefined
  })
  const payments = response.payments
  // ANCHOR_END: list-payments
  return payments
}

const exampleListPaymentsFiltered = async (sdk: BreezSdk): Promise<Payment[]> => {
  // ANCHOR: list-payments-filtered
  // Filter by asset (Bitcoin or Token)
  const assetFilter = new AssetFilter.Token({ tokenIdentifier: 'token_identifier_here' })
  // To filter by Bitcoin instead:
  // const assetFilter = new AssetFilter.Bitcoin()

  const response = await sdk.listPayments({
    // Filter by payment type
    typeFilter: [PaymentType.Send, PaymentType.Receive],
    // Filter by status
    statusFilter: [PaymentStatus.Completed],
    assetFilter,
    // Time range filters
    fromTimestamp: 1704067200n, // Unix timestamp
    toTimestamp: 1735689600n, // Unix timestamp
    // Pagination
    offset: 0,
    limit: 50,
    // Sort order (true = oldest first, false = newest first)
    sortAscending: false,
    sparkHtlcStatusFilter: undefined
  })
  const payments = response.payments
  // ANCHOR_END: list-payments-filtered
  return payments
}
